src/chart/main.c

```diff
@@ -1658,8 +1658,12 @@ static void image_lab_to_xyz(float *image, const int width, const int height)
   for(int y = 0; y < height; y++)
     for(int x = 0; x < width; x++)
     {
-      float *pixel = &image[(x + y * width) * 3];
-      dt_Lab_to_XYZ(pixel, pixel);
+      dt_aligned_pixel_t pixel_lab = { image[(x + y * width) * 3 + 0], image[(x + y * width) * 3 + 1], image[(x + y * width) * 3 + 2]};
```

Comment:
I would try code duplication by using var here:
```
const int i0 = (x + y * width) * 3 + 0;
const int i1 = (x + y * width) * 3 + 1;
const int i2 = (x + y * width) * 3 + 2;
```

And use them here and below.

Reply:
Thanks, that makes the code more readable. I committed the change you proposed.